processing/src/main/java/io/druid/segment/StringDimensionIndexer.java

```diff
@@ -194,19 +222,22 @@ public StringDimensionIndexer(MultiValueHandling multiValueHandling)
     final int oldDictSize = dimLookup.size();
 
     if (dimValues == null) {
-      dimLookup.add(null);
-      encodedDimensionValues = null;
+      if (!dimLookup.contains(null)) {
```

Comment:
Suggested to extract nullId variable and localize if-else (or use ternary)

Reply:
done.